skf/markdown/code_examples/java/4-code_example--Aggregate user contol--.md

```diff
@@ -0,0 +1,282 @@
+            <tr>
+            	<td>Surname : </td>
+            	<td><input type="text" id="surname" name="surname" /></td>
+            </tr>
+            <tr>
+            	<td>Password : </td>
+            	<td><input type="text" id="password" name="password" /></td>
+            </tr>               
+            <tr>
+            	<td><input type="hidden" id="userID" value="2" name="userID" /></td>
+            </tr>         
+        	<tr>
+                <td colspan="2"><input type="submit" /></td>               
+           	</tr>
+        	</table>
+         </form>     
+    </body>
+</html>
+
+
+
+Servlet 
+
+
+the following code snippet can be used in relation with the following servlet snippet
+
+
+public class aggregateControl extends HttpServlet{
+	private static final long serialVersionUID = 1L;
+    public aggregateControl() {
+        super();
+    }
+	
+	final static Logger logger = Logger.getLogger(aggregateControl.class);
+	public String userName;
+	public String password;
+	private String userID;
+
+	protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException 
+	{
+		
+		userName = request.getParameter("name");
+		userID = request.getParameter("userID");
+		Aggregate ag = new Aggregate();
+		ag.setUserName(userName);
+		ag.setUserID(userID);
+	    boolean invalidate_sessions = ag.aggregateControl(0);
+		
+		if (invalidate_sessions == true)
+		{
+			request.getSession().invalidate();			
+		}
+		
+		doGet(request, response);
+		
+		}
+		
+output logs indicating that user has been logged out after many database connections  
+
+[ .... ] 
+
+2016-11-24 11:17:55 INFO  Aggregate:277 - Connection to the database was made successfully   User: ddd  User ID:  2
+2016-11-24 11:18:07 INFO  Aggregate:277 - Connection to the database was made successfully   User: ddd  User ID:  2
+2016-11-24 11:18:15 INFO  Aggregate:277 - Connection to the database was made successfully   User: ddd  User ID:  2
+2016-11-24 11:18:23 INFO  Aggregate:277 - Connection to the database was made successfully   User: ddd  User ID:  2
+2016-11-24 11:18:32 INFO  Aggregate:277 - Connection to the database was made successfully   User: ddd  User ID:  2
+2016-11-24 11:18:39 INFO  Aggregate:277 - Connection to the database was made successfully   User: ddd  User ID:  2
+2016-11-24 11:18:49 INFO  Aggregate:240 -  User account was locked out due to aggregate user control system  User: ddd  Userid ID:  2
+2016-11-24 11:18:49 INFO  Aggregate:277 - Connection to the database was made successfully   User: ddd  User ID:  2
+ 
+ 
+*/
+
+package com.edw;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import javax.naming.Context;
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+import javax.sql.DataSource;
+
+import org.apache.log4j.Logger;
+
+public class NewAggregate {
+	
+	public String userName;
```

Comment:
the username, password and userID fields must be private since there are their setters/getters.